File: back_testing/helpers/Save_data.py
# -*- coding: utf-8 -*-
"""
Created on Thu Jan 30 13:15:09 2025

@author: bravi
"""

import os
from datetime import datetime

def Save_data(data, symbol, start_date=None, end_date=None):
    output_dir = "datos"
    
    # Get the date and actual hour
    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
    
    file_path = os.path.join(output_dir, f"{symbol}_{timestamp}.csv")
    data.to_csv(file_path, index=False)
    print(f"Data saved in {file_path}")

File: back_testing/helpers/Save_data.R
Save_data <- function(data, symbol, start_date = NULL, end_date = NULL) {
  output_dir <- "datos"
  
  # Crear el directorio si no existe
  if (!dir.exists(output_dir)) {
    dir.create(output_dir)
  }
  
  # Obtener la fecha y hora actual
  timestamp <- format(Sys.time(), "%Y%m%d_%H%M%S")
  
  # Crear el nombre del archivo
  file_path <- file.path(output_dir, paste0(symbol, "_", timestamp, ".csv"))
  
  # Guardar los datos en CSV
  write.csv(data, file_path, row.names = FALSE)
  
  message("Data saved in ", file_path)
}
